packages/core/src/lib/jdk/JdkHelper.ts

```diff
@@ -90,18 +91,21 @@ export class JdkHelper {
    * @param {string} jdkPath the path to the jdk.
    */
   static async validatePath(jdkPath: string, currentProcess: NodeJS.Process = process):
-      Promise<Result<string, Error>> {
+      Promise<Result<string, ValidatePathError>> {
     if (!existsSync(jdkPath)) {
-      return Result.error(new Error('jdkPathIsNotCorrect'));
+      return Result.error(new ValidatePathError('The path given isn\'t exist.',
+          'PathIsNotCorrect'));
     };
     const javaHome = this.getJavaHome(jdkPath, currentProcess);
     try {
       const file = await fsPromises.readFile(path.join(javaHome, 'release'), 'utf-8');
       if (file.indexOf('JAVA_VERSION="1.8') < 0) { // Checks if the jdk's version is 8 as needed
-        return Result.error(new Error('jdkIsNotSupported'));
+        return Result.error(new ValidatePathError('The given jdk isn\'t supported.'
+            , 'PathIsNotSupported'));
       }
     } catch (e) {
-      return Result.error(new Error(e.message));
+      return Result.error(new ValidatePathError('Did not find the release file of the jdk.',
+          'PathIsNotCorrect'));
```

Comment:
Could this and PathIsNotSupported be constants in `ValidatePathError`? That way when we're checking the return value of this function, we can use those constants.

Reply:
Do you mean to add those constants in addition to the ErrorCode type which we talked about?

Reply:
Oh, they already are a type (I'd missed `ErrorCode` and thought you were just using these as strings).

Nevermind, code looks good to me.